### ISSUE TITLE
Improve SSUAO to provide more accurate OS information

### ISSUE THREAD

**Issue by JustOff** [MoonchildProductions/UXP#1473]
As discussed [here](https://github.com/MoonchildProductions/Pale-Moon/issues/1733#issuecomment-595282189), we can rewrite ~~the static~~ UA overrides ~~using regexes~~ to more accurately reflect the OS.

**Comment by wolfbeast** [comment]
Moving this to UXP since its platform work and impacts everyone.

**Comment by JustOff** [comment]
So far, I have finished implementing support for the `%OS_SLICE%` macro, as we discussed above, but along the way I also found two more issues regarding SSUAO and dynamic rules update.

The first and most problematic is that all static SSUAO rules always take precedence over those received dynamically, which does not allow using dynamic updating to adjust builtin static rules, but only to add new ones. This can be fixed if we give dynamic rules priority over static rules from the default preference branch, but keep the highest priority for static rules created by the user.

The second is that dynamic rules are always updated at the interval specified in the settings, regardless of restarting the browser. However, given the changes related to the `%OS_SLICE%` macro, it would be better to schedule dynamic updates after a short delay each time the browser starts.

I have already prepared solutions to both of these additional issues and would like to ask how to organize this better. Do I need to create separate issues or can I refer to this one? Do I need to submit three separate pull requests or can I use a single one containing three commits?

**Comment by mattatobin** [comment]
I want to know exactly what you are doing.

**Comment by wolfbeast** [comment]
Please keep this issue focused only on implementing support for the %OS_SLICE% macro!

The other two issues can be looked at together in a single new issue for improving dynamic UA update handling, but don't belong here.
There's a reason I hid several comments because it was already branching off in several things and not remaining focused.

**Comment by JustOff** [comment]
> I want to know exactly what you are doing.

Just as agreed [above](https://github.com/MoonchildProductions/UXP/issues/1473#issuecomment-595348580), details will follow.

> Please keep this issue focused only on implementing support for the %OS_SLICE% macro!

Ok.

**Comment by mattatobin** [comment]
I would like details now please on how you intend to implement this. You can fully explain it or push up a branch somewhere and show me.

**Comment by JustOff** [comment]
Well, [here](https://github.com/JustOff/UXP/commits/_ssuao_os_slice) it is. I am ready to answer your questions.

**Comment by mattatobin** [comment]
Why did you refactor this? Why did you need to change anymore than JUST adding the string replace?

It looks like you are doing more than the scope of what is required.

**Comment by JustOff** [comment]
My work contains the minimum required number of changes to resolve the issue, plus a reasonable number of additional actions aimed at cleaning the code according to #516, removing unused code remaining after your [commit](https://github.com/MoonchildProductions/UXP/commit/bdad2357a1f0d2aca7f1106f3195d1fd929dc1d1#diff-888228427807bd9ad3e953ef1df43f43) according to #1053, and a very small refactoring at the beginning of the one file regarding `Components.utils`. Everything is done the way I do it usually, and, to be honest, I'm a little surprised that I have to explain it.

**Comment by wolfbeast** [comment]
I'm sorry but you obviously don't understand the concept of doing one thing for one issue, here.

All this refactoring has nothing to do with this issue nor is it required to do all the other stuff from other issues in one go (like named function syntax). And then you are completely removing a function from useragentupdates -- which you probably shouldn't even be touching to begin with. Changing ordering of some commands like saving before applying, removing a caching mechanism that is -essential- to minimize performance impact by keeping the UA list in memory, and finally the whole point of this is to get rid of the hard-coded strings for OS.
R- : WTF; Stick to the plan!

**Comment by JustOff** [comment]
> All this refactoring has nothing to do with this issue nor is it required to do all the other stuff from other issues in one go (like named function syntax).

Sorry, but as far as I remember cleaning the code along with making changes was always welcomed here, and at least it was never explicitly forbidden. I have no problem following the rules when they are clearly defined, and I always try to ask when doubts arise. Well, I think now I understand you.

>  And then you are completely removing a function from

That was a code cleanup part which I was going to explain in PR, as I usually did, but ok, this case is closed, I will never try to clean up again.

> useragentupdates -- which you probably shouldn't even be touching to begin with. Changing ordering of some commands like saving before applying, 

The other changes apart from removing unused function are essential to the main task.

> removing a caching mechanism that is -essential- to minimize performance impact by keeping the UA list in memory, 

`UserAgentOverrides.jsm` uses several layers of caching and believe me, the one I removed is redundant and prevents the browser from responding correctly to SSUAO updates. It is involved only at the stage of rebuilding rules, which is extremely rare and does not affect the performance of network requests.

> and finally the whole point of this is to get rid of the hard-coded strings for OS... and yet there they are. I told you I don't want them.

Perhaps we did not understand each other. I [meant](https://github.com/MoonchildProductions/UXP/issues/1473#issuecomment-595358455) to use them as a _fallback_ and there were [no objections](https://github.com/MoonchildProductions/UXP/issues/1473#issuecomment-595358811). What other fallback method did you mean?

> Stick to the plan!

Don't get me wrong, but I took some time to figure out the issue, and you tell me that I removed the important cache, which I am sure it is not. Now I simply don't understand where and how to move on.

**Comment by mattatobin** [comment]
You should work things out ahead of time and plan what you want to accomplish and how best to go about that. Just because people like your self and @janekptacijarabaci got away with that in the past has no barring on today.

Today, we want those who want to do something to explain it to us. We also want unrelated changes to not mask the changes specifically desired to accomplish a task. If you want to add clean up and such as part of your work then it can be permitted on a case-by-case basis but we need to know that is your intention before hand and it needs to happen in separate commits if not separate pull requests.

BTW, while it may not always seem that I personally do not adhere to these things when I do work, you should know that very little of what I do is done without someone, mainly @wolfbeast, knowing about it in far more detail than he would probably prefer or is required. Just because it doesn't expressly happen on GitHub doesn't mean it doesn't happen. I simply have alternative means of communication. You do not.

The bottom line is we are all answerable to @wolfbeast but you are also loosely answerable to me as well when things impact more than one application which is basically all of UXP. This is true for every contributor so that we won't be in situations like this or those caused by @janekptacijarabaci and others.

I am sorry that the structure and guidelines were not previously made clear in absolute terms but not all of them are so one has to be flexible and adapt to changing circumstances. This would be one of them.

If you would like, we can close all these issues and create new ones to wipe the slate clean and coordinate, discuss, and focus what needs to be done and why and how. That is up to you, though.

**Comment by JustOff** [comment]
> You should work things out ahead of time and plan what you want to accomplish and how best to go about that. Just because people like your self and @janekptacijarabaci got away with that in the past has no barring on today.

I understand your frustration about things not always going the way they should, but I always try to describe as much as possible what and how I do, listen to the comments and make the necessary adjustments. So please do not generalize and do not compare me with those who could not ensure the required level of communication.

> Today, we want those who want to do something to explain it to us. We also want unrelated changes to not mask the changes specifically desired to accomplish a task. If you want to add clean up and such as part of your work then it can be permitted on a case-by-case basis but we need to know that is your intention before hand and it needs to happen in separate commits if not separate pull requests.

Let me repeat that I have no objection to this, and I will adhere to the requirements concerning unrelated changes, as they have now been explicitly stated.

> BTW, while it may not always seem that I personally do not adhere to these things when I do work, you should know that very little of what I do is done without someone, mainly @wolfbeast, knowing about it in far more detail than he would probably prefer or is required. Just because it doesn't expressly happen on GitHub doesn't mean it doesn't happen. I simply have alternative means of communication. You do not.

I hope that in my case we will be able to continue using GitHub as a single place of communication without compromising on its quality. I also think that if others also used this method more often, it would benefit everyone.

> The bottom line is we are all answerable to @wolfbeast but you are also loosely answerable to me as well when things impact more than one application which is basically all of UXP.

I understand and have no problem with this.

> I am sorry that the structure and guidelines were not previously made clear in absolute terms but not all of them are so one has to be flexible and adapt to changing circumstances. This would be one of them.

You have nothing to apologize for, I'm always ready to adapt to any reasonable requirements.

> If you would like, we can close all these issues and create new ones to wipe the slate clean and coordinate, discuss, and focus what needs to be done including why and how. That is up to you, though.

I don't mind creating new issues if you want, although I don't quite understand how this will help. My next step, in any case, will be to remove all unrelated changes, but in order to move on we need to deal with the actual side of the issue. To do this, I would like to get answers to the questions that I asked @wolfbeast in the [previous comment](https://github.com/MoonchildProductions/UXP/issues/1473#issuecomment-595968218).

**Comment by wolfbeast** [comment]
I don't see any specific questions aside from what I meant with a fallback.
Fallback should be whatever the (unmodified) UA is; and in fact that should probably just be used anyway instead of "after override". There's no need to create all sorts of elaborate things around it -- `%OS_SLICE%` should simply be substituted by what the application constructs as standard for its OS component of the useragent string.
All this issue needs (and I mean _all_) is:
1. get/construct the OS slice as-generated by the networking component
2. substitute `%OS_SLICE%` with (1) while parsing/creating the SSUAOs

If this has as a prerequisite that the `map` with previously constructed UAs not be used or needs to be used differently, then that should be a _different_ issue **blocking this one**. In that issue there needs to be a clear description what the problem is with it. I still think that this is essential for performance, so I need to understand why it is not (and in that case, why would Mozilla have created it in the first place?)...

As a general note: previously I've allowed for things to be "rolled in" to a PR with an explanation on occasion, but I _really_ want to avoid that because it makes debugging harder than it has to be, and makes it practically impossible to find commits where specific unrelated things were changed as part of the solution for something else. So anything not directly related to creating this slice and substituting it in SSUAOs _doesn't_ belong here. Cleanup belongs in a cleanup issue. Removing the cache belongs in its own issue as stated, etc. etc.

**Comment by JustOff** [comment]
I'm sorry, but some real life issues distracted me a little. I will try to return and continue within a few days.

**Comment by mattatobin** [comment]
>real life

**Comment by JustOff** [comment]
First of all, after further analysis, I must admit that I really missed the simple way to get the default OS slice without the ugly hardcoding of this value at compile time. Now I have found such a method and will use it in the updated version of the patch.

Further, regarding the remaining changes that caused the most questions. In the previous discussion, you [wrote](https://github.com/MoonchildProductions/Pale-Moon/issues/1733#issuecomment-595256168) "In fact, I'm sure if someone explicitly overrides useragents generically, they would want whatever OS they chose to override with to be applied in these cases as well." and I took it as a _direct directive_ to take this nuance into account when working on this pull request. In fact, the implementation of this requirement took the most of my time and most of the changes were also made due to this.

And in addition, all this happened against the background of hardening requirements for pull requests, which I learned only after the fact. Not that I'm complaining, but please understand my confusion when all this fell on me at the same time, while I was sure that I was acting according to a well-agreed plan.

Well, fortunately now, after your last comment, everything became almost clear, however, to be sure, I want to clarify a few more things:

- > your regex should not be case insensitive and it should be global to replace any occurrences

  I can agree not to make regex case insensitive (although I think it could be useful), but why might a global replacement be necessary if `%OS_SLICE%` can only occur once in SSUAO?

- Which coding style is correct (if any):

```
const OS_SLICE = Cc["@mozilla.org/network/protocol;1?name=http"]
                   .getService(Ci.nsIHttpProtocolHandler)
                   .oscpu + ";";
```
or
```
const OS_SLICE = Cc["@mozilla.org/network/protocol;1?name=http"]
                   .getService(Ci.nsIHttpProtocolHandler)
                   .oscpu +
                   ";";
```

- Do you still consider it important to take global user agent override into account when resolving `%OS_SLICE%` macro, as you [asked](https://github.com/MoonchildProductions/Pale-Moon/issues/1733#issuecomment-595256168) before, i.e., should I open a new issue about this or is this requirement canceled now?

**Comment by wolfbeast** [comment]
1. any occurrences of the placeholder should be replaced, if present, not just the first one. That's the whole point of using these, to have them _always_ replaced.
2. the first style is correct.
3. It was a side remark that _if_ it happens to take the general override into account (i.e. using that UA string), that in that case it's people's own doing if it's not accurate, but it was never meant as a directive t make that happen. I'd rather not have this behavior ;-)

**Comment by mattatobin** [comment]
That is more like it. Thank you focusing your patch and for the research put into this.

When you conform `uaoverrides` to this in `Pale Moon` and `Basilisk`, please reference this issue.

**Comment by JustOff** [comment]
Perhaps I should better wait until the platform pointer is updated to include #1476 in both repos?

**Comment by mattatobin** [comment]
You can still do the PRs and @wolfbeast can update the commit pointer when he merges them in.

**Comment by JustOff** [comment]
I think we are finally done here, isn't it?

**Comment by JustOff** [comment]
Thanks all once again for being patient and respectful.

**Comment by JustOff** [comment]
It seems due to the lack of testing and feedback on Linux I messed the things up.

**Comment by JustOff** [comment]
I hope we can hotfix this using dynamic SSUAO, I'll try to check it right now.

**Comment by mattatobin** [comment]
And you didn't test this? WHY?!

**Comment by JustOff** [comment]
> And you didn't test this?

I did a short test, but didn't notice there is no `X11; `. Maybe I'm blind ;-/

**Comment by mattatobin** [comment]
Well PR a fix and @trav90 will simply have to do an out of band `.1` I guess.

**Comment by wolfbeast** [comment]
Yeah an out-of-band 28.9.0.1 will be needed.
I'll check if dynamic updates can help but I'm not sure if that'll be faster or not. We don't check that often for updates to it.

**Comment by JustOff** [comment]
I'll try to come up with a fix ASAP, but for a time being I suggest to update the rules using dynamic SSUAO.

Here is DynSSUAO to test:

```
{
  "addons.mozilla.org": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:4.5) Goanna/20170101 PaleMoon/28.9.0",
  "live.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:52.9) Gecko/20100101 Firefox/52.9 (Pale Moon)",
  "msn.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:52.9) Gecko/20100101 Firefox/52.9 (Pale Moon)",
  "bing.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:52.9) Gecko/20100101 Firefox/52.9 (Pale Moon)",
  "outlook.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:52.9) Gecko/20100101 Firefox/52.9 (Pale Moon)",
  "web.de": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:52.9) Gecko/20100101 Firefox/52.9 (Pale Moon)",
  "aol.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:52.9) Gecko/20100101 Firefox/52.9 (Pale Moon)",
  "calendar.yahoo.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:52.9) Gecko/20100101 Firefox/52.9 (Pale Moon)",
  "google.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:71.0) Gecko/20100101 Firefox/71.0 PaleMoon/28.9.0",
  "googlevideos.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:38.9) Gecko/20100101 Goanna/4.5 Firefox/38.9 PaleMoon/28.9.0",
  "gstatic.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:71.0) Gecko/20100101 Firefox/71.0 PaleMoon/28.9.0",
  "yahoo.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:99.9) Gecko/20100101 Firefox/99.9 (Pale Moon)",
  "youtube.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:60.0) Gecko/20100101 Firefox/60.0 PaleMoon/28.9.0",
  "gaming.youtube.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:42.0) Gecko/20100101 Firefox/42.0",
  "dropbox.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:99.9) Gecko/20100101 Firefox/99.9 (Pale Moon)",
  "players.brightcove.net","Mozilla/5.0 (Windows NT 6.1; Trident/7.0; rv:11.0) like Gecko",
  "facebook.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:99.9) Gecko/20100101 Firefox/99.9 (Pale Moon)",
  "fbcdn.net": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:99.9) Gecko/20100101 Firefox/99.9 (Pale Moon)",
  "chase.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:52.9) Gecko/20100101 Firefox/52.9",
  "www.amazon.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:45.9) Gecko/20100101 Firefox/45.9 (Pale Moon)",
  "soundcloud.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:4.5) Goanna/20170101 PaleMoon/28.9.0",
  "dailymotion.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:52.0) Gecko/20100101 Firefox/52.0",
  "deviantart.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:4.5) Goanna/20170101 PaleMoon/28.9.0",
  "deviantart.net": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:4.5) Goanna/20170101 PaleMoon/28.9.0",
  "altibox.dk": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:4.5) Goanna/20170101 PaleMoon/28.9.0",
  "altibox.no": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:4.5) Goanna/20170101 PaleMoon/28.9.0",
  "firefox.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:4.5) Goanna/20170101 PaleMoon/28.9.0",
  "mozilla.org": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:4.5) Goanna/20170101 PaleMoon/28.9.0",
  "mozilla.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:4.5) Goanna/20170101 PaleMoon/28.9.0",
  "github.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:4.5) Goanna/20170101 PaleMoon/28.9.0",
  "humblebundle.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:52.9) Gecko/20100101 Firefox/52.9 (Pale Moon)",
  "privat24.ua": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:38.0) Gecko/20100101 Firefox/38.0",
  "citi.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:57.0) Gecko/20100101 Firefox/57.0 (Pale Moon)",
  "netflix.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:45.9) Gecko/20100101 Firefox/45.9",
  "netflximg.net": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:45.9) Gecko/20100101 Firefox/45.9",
  "web.whatsapp.com","Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.77 Safari/537.36",
  "hitbox.tv": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:52.9) Gecko/20100101 Firefox/52.9",
  "yuku.com": "Mozilla\/5.0 \((.+?) rv:.*#Mozilla/5.0 ($1 rv:52.9) Gecko/20100101 Firefox/52.9 PaleMoon/28.9.0"
}
```

**Comment by mattatobin** [comment]
With what infra?

**Comment by JustOff** [comment]
Damn, I'm confused, it doesn't work as it should. I'm trying to figure it out right now. I will let you know as soon as there is something to say. And sorry, I seem to forget to say sorry. I'm really sorry.

**Comment by wolfbeast** [comment]
It seems people on 32-bit windows are also affected?
Dynamic updates seem to work as-intended though...

**Comment by wolfbeast** [comment]
just tested win32 28.9.0 portable. pasted the yahoo string in general.useragent.override.useragentstring.com and it spat the correct thing back at me:
`Mozilla/5.0 (Windows NT 6.1; WOW64; rv:99.9) Gecko/20100101 Firefox/99.9 (Pale Moon)`
So win32 shouldn't be affected.

**Comment by JustOff** [comment]
> It seems people on 32-bit windows are also affected?

I will check it too. I think it makes sense to add all domains to dynSSUAO for now. I want to double-check everything before proposing a general solution.

**Comment by wolfbeast** [comment]
I dropped the dynamic overrides in for both channels for now.

**Comment by JustOff** [comment]
So, as far as I can see, we have two options:
 - Try to replicate the `OS_SLICE` [building logic](https://github.com/MoonchildProductions/UXP/blob/master/netwerk/protocol/http/nsHttpHandler.cpp#L711-L731) from `nsHttpHandler.cpp` in `UserAgentOverrides.jsm`
 - Add a new `osslice` property to `nsIHttpProtocolHandler`, which will be calculated directly in `nsHttpHandler.cpp` along with `mUserAgent`

The latter seems to me preferable, what do you say?

**Comment by trav90** [comment]
Are we sure that this is Linux-specific? I tested on two different systems (one 64-bit, one 32-bit) and didn't notice any issues (still had the "new" YouTube, design, etc.).

**Comment by mattatobin** [comment]
Rabbit hole.

**Comment by JustOff** [comment]
Well, while none of the options that I proposed has been approved, I will prepare a patch that only changes `UserAgentOverrides.jsm` and does not touch `nsIHttpProtocolHandler`. At least it will be safer for a point release.

**Comment by wolfbeast** [comment]
I don't understand why some people have issues with this and others don't. We can try to band-aid this but right now I'm actually in favor of just backing this out and going back to static O.S. slices until we can properly figure out what's going wrong here, and why.

**Comment by JustOff** [comment]
Well, the patch is ready and I'm currently testing it, but you decide.

**Comment by wolfbeast** [comment]
I think it wrong here to scramble for a solution to shove into -release for something the cause isn't clear of. We can look at this patch on a dev branch, but I'd rather return -release to a known-good configuration right now.

**Comment by wolfbeast** [comment]
I backed out both #1473 and #1482 from the release branch because i think part of the problem is the change in `Init` behavior. Having a dirty/uninitialized cache can cause issues after all.
I backed out the front-end changes on MoonchildProductions/Pale-Moon reverting the config prefs back to `@OS_SLICE@` with hard-coded strings, and tagged the result which should be good to go for build and update.

**Comment by JustOff** [comment]
I believe the only problem is with the incorrect `OS_SLICE` for Linux, but the release is definitely not the place to check, since you have these doubts and especially considering this has already affected users.

**Comment by wolfbeast** [comment]
It has also affected Windows users so there's something more going on.

**Comment by trav90** [comment]
> I backed out both #1473 and #1482 ... and tagged the result which should be good to go for build and update.

@wolfbeast I've got updated Linux tarballs ready for release.

**Comment by wolfbeast** [comment]
I'm putting the final touches on Windows to upload & inform mirrors etc.
Should be ready to publish in 30 or so.

**Comment by JustOff** [comment]
> It has also affected Windows users so there's something more going on.

From what I see on the forum, some users still have the same problems with 28.9.0.1, so it seems the reason is not #1473 or #1482, but something else.

**Comment by JustOff** [comment]
And most likely, with the exception of incorrect building `%OS_SLICE%` on non-Windows platforms, everything else was caused by #1495.

**Comment by wolfbeast** [comment]
You're probably right. But hindsight is always perfect ;)
We can just re-land this if it doesn't turn out to be problematic in light of the init issue.

**Comment by wolfbeast** [comment]
It's landed and enabled in the current unstable. feel free to verify. I'd like to close this.

**Comment by JustOff** [comment]
Please remind me if there are any unstable builds for Linux available? And if not, could someone make a fresh one?

**Comment by wolfbeast** [comment]
Linux site has unstables

**Comment by JustOff** [comment]
Oops, I always forget the Linux site has its own menu.

I can confirm that `OS_SLICE` on Linux is now built correctly.

**Comment by mattatobin** [comment]
I can build on the Linux. Only takes about ~15 minutes.

**Comment by JustOff** [comment]
Thanks, this is no longer necessary. One could still check on OS X or Solaris, but if everything is ok on Linux, then it should also be fine on other non-Windows platforms.
